habitat-lab/habitat/tasks/rearrange/rearrange_task.py

```diff
@@ -243,6 +248,10 @@ def reset(self, episode: Episode, fetch_observations: bool = True):
             self._is_episode_active = True
 
             if self._should_place_articulated_agent:
+                if self._fixed_agent_starting_pose:
```

Comment:
Are you happy with this implementation? I see two old comments that suggest this wasn't quite working correctly yet:

From your PR description:
>slightly different agents positions even though we force them to be fixed by adding "fixed_agent_starting_pose" config argument (for the context if "fixed_agent_starting_pose" is in config we fix numpy and sim.pathfinder seeds). probably because float numbers representation? This causes test to fail sometimes.

From [this Slack thread](https://cvmlp.slack.com/archives/C0460NTKM4G/p1698675347202439?thread_ts=1698438249.886569&cid=C0460NTKM4G):
>So, I would expect that setting env.seed(my_seed) would be enough to have deterministic behaviour, but it is not, agent positions are set to different positions every time. Is it working as expected [@Andrew Szot](https://cvmlp.slack.com/team/U011205HNNT) [@Alex Clegg](https://cvmlp.slack.com/team/UKZ187YA3)?